server/src/instant/db/datalog.clj

```diff
@@ -808,17 +820,23 @@
     [:and [:= :app-id app-id]
           [:= :value [:cast \"25\" :jsonb]]]
   "
-  [app-id {:keys [idx] :as named-pattern} additional-clauses]
-  (list*
-   :and
-   [:= :app-id app-id]
-   [:= (idx-key idx) :true]
-   (concat (->> named-pattern
-                constant-components
-                (map (fn [[component-type v]]
-                       (constant->where-part idx app-id component-type v))))
-           (function-clauses app-id named-pattern)
-           (patch-values-for-av-index (idx-key idx) additional-clauses))))
+  [app-id triples-alias {:keys [idx] :as named-pattern} additional-clauses]
+  (let [attr-ids-where (when-let [a (:a named-pattern)]
+                         (when (named-constant? a)
+                           (constant->where-part idx app-id :a a)))
+        rest-wheres (concat (->> (dissoc named-pattern :a)
+                                 constant-components
+                                 (map (fn [[component-type v]]
+                                        (constant->where-part idx app-id component-type v))))
+                            (function-clauses app-id triples-alias named-pattern)
+                            (patch-values-for-av-index (idx-key idx) additional-clauses))]
+    (list*
```

Comment:
This change causes some problems, so I'm going to update this to only leave out the index if we're doing the count query.